test/support/data_case.ex

```diff
@@ -22,4 +22,19 @@ defmodule Beacon.DataCase do
 
     :ok
   end
+
+  @doc """
+  Helper function for setting up data source.
+
+  ### Examples
+
+      with_data_source(fn ->
+        # your test code in here...
+      end)
+  """
+  def with_data_source(datasource \\ DummyApp.BeaconDataSource, assert_func) when is_function(assert_func) do
+    Application.put_env(:beacon, :data_source, datasource)
+    assert_func.()
+    Application.delete_env(:beacon, :data_source)
```

Comment:
Can we look at using Mox for the BeaconDataSource? I would like to avoid swapping env vars on the fly as it can get messy, and prevents us from using async tests.

Reply:
True. I was trying to be simple here but forgot the messing with env prevents async tests. I will update it soon

Reply:
actually, we don't need mox right now since I'm using the same datasource for both tests. We might need in the future once we get more tests that needs different data source, but we can live without it for the time being if that makes sense.